sway-lib-std/src/tx.sw

```diff
@@ -232,6 +233,29 @@ pub fn tx_script_bytecode<T>() -> T {
     read::<T>(tx_script_start_pointer())
 }
 
+/// Get the hash of the script bytecode.
+/// Reverts if not a transaction-script
+pub fn tx_script_bytecode_hash() -> b256 {
+    let type = tx_type();
+    match type {
+        Transaction::Script => {
+            // Get the script memory details
+            let mut result_buffer: b256 = ZERO_B256;
+            let script_length = __gtf::<u64>(0, GTF_SCRIPT_SCRIPT_LENGTH);
+            let script_ptr = __gtf::<u64>(0, GTF_SCRIPT_SCRIPT);
+            
+            // Run the hash opcode for the script in memory
+            asm(hash: result_buffer, ptr: script_ptr, len: script_length) {
+                s256 hash ptr len;
```

Comment:
I guess there is no way to use the `sha256` library here :/

Reply:
Why is that?

Reply:
Mainly because all we have is a pointer and a length and not a concrete value that we can just pass to `sha256`. If we somehow can be build an array of bytes (`u8`s) out of this, then we can just use `sha256` directly, but I'm worried that's not possible just yet (and `Vec` doesn't help here because we'd only be hashing the struct itself and not the heap data).

Reply:
@mohammadfawaz is correct. It's not currently possible to get an array of `u8`s through a simple compiler supported casting or anything. You could get an array of 64 bit "bytes", zero-padded to `u64` length, or the whole data rounded to the nearest `u64` with padded zeros at the end, but then this messes up the computed hash (and also is memory intensive). Since this is being used for bridging in a predicate that is likely to be used often, I think it makes sense to have an ultra optimized assembly defined function rather than try and fix current gaps.